pkg/devcontainer/setup/setup.go

```diff
@@ -206,6 +212,20 @@ func PatchEtcEnvironment(mergedConfig *config.MergedDevContainerConfig, log log.
 	return nil
 }
 
+func ChownAgentSock(setupInfo *config.Result, log log.Logger) error {
+	user := config.GetRemoteUser(setupInfo)
+
+	agentSockFile := os.Getenv("SSH_AUTH_SOCK")
+	if agentSockFile != "" {
+		err := copy2.ChownR(filepath.Dir(agentSockFile), user)
```

Comment:
why do we need to `chown` the whole directory here?

Reply:
I had tried that initially.
So the way the agent sock parent directories are created is like below. Notice how root is the owner of the directory and has full permissions while group and others have no permission whatsoever.
```
$ ls -l /tmp/
drwx------ 2 root root  4096 Oct 26 07:38 auth-agent3289311155
drwx------ 2 root root  4096 Oct 26 07:04 auth-agent961926320
```

Now, if I just change the owner of the sock file, I get permissions on the file, but still no access on the parent directory. Due to this, we get _: Permission denied_ as it can't traverse the directory chain.

There are 2 options to make this work:
1. Give the read and execute permissions to the _others_ group on this directory and the sock file.
2. Change the owner of the directory and the file to the remoteUser.

I have used option 2.

Reply:
I see, thanks for the explanation